R/bploop.R

```diff
@@ -268,20 +261,26 @@ bploop.iterate <-
     on.exit(progress$term(), TRUE)
     progress$init()
 
-    BPRNGSEED <- .rng_seeds_by_task(BPPARAM, TRUE, 1L)[[1]]
-
+    ARGFUN <- function(X, seed)
+        c(
+            list(X=X), list(FUN=FUN), ARGS,
+            list(BPRNGSEED = seed)
+        )
     ## initial load
     for (i in seq_len(workers)) {
         value <- ITER()
-        if (is.null(value)) {
+        if (inherits(value, ".rng_bploop_iter")) {
+            seed <- .rng_iterate_substream(seed, value)
+            value <- ITER()
```

Comment:
It looks like `ITER()` is called twice in the for loop?

Reply:
Yes, we need to make the iterator works with BPREDO, so I introduce a new class `.rng_bploop_iter` witch is merely designed to inform `bploop` to iterate the rng substream a few times. After the iteration, we should retrieve the next value from the iterator, which brings L274. There should be no two consecutive `.rng_bploop_iter` in the iterator, so we do not have to check the value class again.

Reply:
Can you add a clarifying comment to the code?

Reply:
No problem, I have updated the comments.